Wordle.Api/Services/WordService.cs

```diff
@@ -0,0 +1,97 @@
+﻿using Wordle.Api.Data;
+using System.Linq;
+using static Wordle.Api.Data.Game;
+using Microsoft.EntityFrameworkCore;
+using System.Collections.Concurrent;
+using Wordle.Api.Dtos;
+
+namespace Wordle.Api.Services;
+
+public class WordService
+{
+    private readonly AppDbContext _context;
+
+    public WordService(AppDbContext context)
+    {
+        _context = context;
+    }
+
+    public IEnumerable<Word> GetWordList(int wordsPerPage, int pageNum, string wordFilter)
+    {
+
+        if (String.IsNullOrWhiteSpace(wordFilter))
+        {
+            var result = _context.Words
+                .Skip((pageNum - 1) * wordsPerPage)
+                .Take(wordsPerPage)
+                .OrderBy(x => x.Value);
+            return result;
+        }
+        else
+        {
+            var result = _context.Words
+                .Where(x => x.Value.StartsWith(wordFilter) && x.Active)
+                .Skip((pageNum - 1) * wordsPerPage)
+                .Take(wordsPerPage)
+                .OrderBy(x => x.Value);
+            return result;
+        }
+    }
+
+    public int GetTotalWordCount(string wordFilter)
+    {
+        var result = _context.Words.Where(x => x.Value.StartsWith(wordFilter)).Count();
```

Comment:
this function might be causing your app to run slower. not sure why you need a total count that is separate.